modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/cell_renderers/DateTimeRenderer.js

```diff
@@ -10,15 +10,24 @@ function DateTimeRenderer({options, value}) {
 		return null;
 	}
 
-	const locale = themeDisplay.getLanguageId().replaceAll('_', '-');
-	const dateOptions = options?.format || {
-		day: 'numeric',
-		hour: 'numeric',
-		minute: 'numeric',
-		month: 'short',
-		second: 'numeric',
-		year: 'numeric',
+	const locale = Liferay.ThemeDisplay.getBCP47LanguageId();
+	const dateOptions = {
+		day: options?.format?.day || 'numeric',
+		hour: options?.format?.hour || 'numeric',
+		minute: options?.format?.minute || 'numeric',
+		month: options?.format?.month || 'short',
+		second: options?.format?.second || 'numeric',
+		timeZone: options?.format?.timeZone,
+		year: options?.format?.year || 'numeric',
 	};
+
+	if (
+		!dateOptions.timeZone &&
+		/^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2,3}Z$/.test(value)
+	) {
+		dateOptions.timeZone = Liferay.ThemeDisplay.getTimeZone();
+	}
```

Comment:
It's fine if we want to limit expected format of date string, but this seems inadequate:
- It's not just time zones application, `new Date(value)` on L32 is problematic (see https://github.com/liferay-frontend/liferay-portal/pull/1722#discussion_r763108253) . Why is ISO format specifically relevant to this case?
- We don't make any expected format check in `DateRenderer`. It will just crash if date is not ISO format.

Reply:
Yep, I see this renderer might produce unexpected results from end user perspective for certain combination of string date values. I'd say, however, that this "issue" is prior to this code and I'm not making any attempt at fixing it.

ISO format is important because we must expect Liferay REST applications to return dates in that format according to product management. So, basically I'm making the time zone conversion to the target user only if all data is available and the timezone has not been specified in the options.

This way, conversion takes place on a controlled set of cases. If date comes in other formats, then we are preserving previous behavior.

Tbh I don't think this should be the "final" version of this renderer, and that's why I added some conditions to the conversion, in a sort of defensive measure.

We shall open a discussion about how robust/capable this renderer must be to handle a variety of cases. We haven't done this so far, thus there's a lack of clarity about how far should we go.

Still, I think this represents a reasonable balance to handle the issue at hand

Reply:
The check will not cause any issues, but I don't think there is any benefit from it. If date is in a wrong, non-ISO format, we can still apply timezone shift. Time zone setting ends up in `dateOptions` object, used by `Intl.DateTimeFormat`. That will always work fine. The problematic part about ISO format is `new Date(value)` on L32, and it may break in funny ways if format is not ISO. This was an issue before this PR, and will be after it. My point is, I don't think we don't need a format check specific for time zones. If we are adding a format check, it should be the first thing in renderer, failing somehow or returning raw value.